src/c++/library/http_client.cc

```diff
@@ -458,6 +468,13 @@ HttpInferRequest::PrepareRequestJson(
       }
       io_json.Add("shape", std::move(shape_json));
 
+      bool json_data{false};
+      if (std::shared_ptr<InferInput::HttpOptions> options =
+              std::dynamic_pointer_cast<InferInput::HttpOptions>(
```

Comment:
Is it possible to move the assignment out of the condition?

Reply:
Irrelevant now 👍